Docs: Recipe for styling with Emotion (#17351)
gatsbyjs/gatsby: docs/docs/recipes.md

- More on [importing assets into files](/docs/importing-assets-into-files/)

### Using Emotion

> [Emotion](https://emotion.sh) is a powerful CSS-in-JS library that suports both inline CSS styles and styled components.

Reviewer: ```suggestion
[Emotion](https://emotion.sh) is a powerful CSS-in-JS library that supports both inline CSS styles and styled components.
```

#### Prerequisites

- A [Gatsby site](/docs/quick-start)

#### Directions

> Install the [Gatsby Emotion plugin](https://www.gatsbyjs.org/packages/gatsby-plugin-emotion/) and Emotion packages.

Reviewer: ```suggestion
1. Install the [Gatsby Emotion plugin](/packages/gatsby-plugin-emotion/) and Emotion packages.
```

```shell
npm install --save gatsby-plugin-emotion @emotion/core @emotion/styled
```

> Add the `gatsby-plugin-emotion` plugin to your `gatsby-config.js` file:

Reviewer: ```suggestion
2. Add the `gatsby-plugin-emotion` plugin to your `gatsby-config.js` file:
```

```javascript:title=gatsby-config.js
plugins: [
  `gatsby-plugin-emotion`,
],
```

> Create a page in your gatsby site at `/src/pages/emotion-sample.js`. Import the Emotion CSS and styled packages:

Reviewer: ```suggestion
3. Create a page in your gatsby site at `/src/pages/emotion-sample.js`. Import the Emotion CSS and styled packages:
```

> `````jsx:title=/src/pages/emotion-sample.js

Reviewer 2: too much backticks...

```suggestion
```jsx:title=/src/pages/emotion-sample.js
```

import React from "react"
import { css } from "@emotion/core"
import styled from "@emotion/styled"

export default () => (
  <div>
    <p>This page is using Emotion.</p>
  </div>
)
```

> You can use Emotion's `css` prop to add [Emotion object styles](https://emotion.sh/docs/object-styles) to any element:

Reviewer: ```suggestion
4. You can use Emotion's `css` prop to add [Emotion object styles](https://emotion.sh/docs/object-styles) to any element:
```

> ````jsx:title=/src/pages/emotion-sample.js

Reviewer 2: too much backticks...

```suggestion
```jsx:title=/src/pages/emotion-sample.js
```

export default () => (
  <div>
    <p
      css={{
        background: "pink",
        color: "blue",
      }}
    >
      This page is using Emotion.
    </p>
  </div>
)
```

> To use [styled components](https://emotion.sh/docs/styled), define them using Emotion's `styled` function.

Reviewer: ```suggestion
5. To use [styled components](https://emotion.sh/docs/styled), define them using Emotion's `styled` function.
```

> ````jsx:title=/src/pages/emotion-sample.js

Reviewer 2: too much backticks...

```suggestion
```jsx:title=/src/pages/emotion-sample.js
```

- More on [importing assets into files](/docs/importing-assets-into-files/)

### Using Emotion

> [Emotion](https://emotion.sh) is a powerful CSS-in-JS library that supports both inline CSS styles and styled components.

Reviewer: ```suggestion
[Emotion](https://emotion.sh) is a powerful CSS-in-JS library that supports both inline CSS styles and styled components. You can use each styling feature individually or together in the same file.
```

#### Prerequisites

- A [Gatsby site](/docs/quick-start)

#### Directions

1. Install the [Gatsby Emotion plugin](/packages/gatsby-plugin-emotion/) and Emotion packages.

```shell
npm install --save gatsby-plugin-emotion @emotion/core @emotion/styled
```

2. Add the `gatsby-plugin-emotion` plugin to your `gatsby-config.js` file:

```javascript:title=gatsby-config.js

> plugins: [

Reviewer 3: I think it's better to show the complete code example so that people can copy a working snippet.

```
module.exports = {
  plugins: [`gatsby-plugin-emotion`],
}
```

  `gatsby-plugin-emotion`,
],
```

> 3. Create a page in your gatsby site at `/src/pages/emotion-sample.js`. Import the Emotion CSS and styled packages:

Reviewer 3: ```suggestion
3. Create a page in your gatsby site at `src/pages/emotion-sample.js`. Import the Emotion CSS and styled packages:
```

Removing the leading `/` - other recipes/docs don't do that

> 3. Create a page in your gatsby site at `src/pages/emotion-sample.js`. Import the Emotion CSS and styled packages:

Reviewer: I think it would be more clear (and slightly shorter) if we move the imports to each item. That way people don't think they have to necessarily be used together.

> ```jsx:title=/src/pages/emotion-sample.js

Reviewer 3: ```suggestion
```jsx:title=src/pages/emotion-sample.js
```

import React from "react"
import { css } from "@emotion/core"
import styled from "@emotion/styled"

export default () => (
  <div>
    <p>This page is using Emotion.</p>
  </div>
)
```

> 4. You can use Emotion's `css` prop to add [Emotion object styles](https://emotion.sh/docs/object-styles) to any element:

Reviewer: ```suggestion
3. After importing Emotion's `css` core package, you can use the `css` prop to add [Emotion object styles](https://emotion.sh/docs/object-styles) to any element inside a component:
```

> ```jsx:title=/src/pages/emotion-sample.js

Reviewer 3: ```suggestion
```jsx:title=src/pages/emotion-sample.js
```

> ```jsx:title=src/pages/emotion-sample.js

Reviewer: ```suggestion
```jsx:title=src/pages/emotion-sample.js
import React from "react"
import { css } from "@emotion/core"

```

export default () => (
  <div>
    <p
      css={{
        background: "pink",
        color: "blue",
      }}
    >
      This page is using Emotion.
    </p>
  </div>
)
```

> 5. To use [styled components](https://emotion.sh/docs/styled), define them using Emotion's `styled` function.

Reviewer: ```suggestion
4. To use Emotion's [styled components](https://emotion.sh/docs/styled), import the package and define them using the `styled` function.
```

> ```jsx:title=/src/pages/emotion-sample.js

Reviewer 3: ```suggestion
```jsx:title=src/pages/emotion-sample.js
```

> ```jsx:title=src/pages/emotion-sample.js

Reviewer: ```suggestion
```jsx:title=src/pages/emotion-sample.js
import React from "react"
import styled from "@emotion/styled"

```

- More on [importing assets into files](/docs/importing-assets-into-files/)

### Using Emotion

[Emotion](https://emotion.sh) is a powerful CSS-in-JS library that supports both inline CSS styles and styled components. You can use each styling feature individually or together in the same file.

#### Prerequisites

- A [Gatsby site](/docs/quick-start)

#### Directions

1. Install the [Gatsby Emotion plugin](/packages/gatsby-plugin-emotion/) and Emotion packages.

```shell
npm install --save gatsby-plugin-emotion @emotion/core @emotion/styled
```

2. Add the `gatsby-plugin-emotion` plugin to your `gatsby-config.js` file:

```javascript:title=gatsby-config.js
module.exports = {
  plugins: [`gatsby-plugin-emotion`],
}
```

> 3. Create a page in your gatsby site at `src/pages/emotion-sample.js`. Import the Emotion CSS and styled packages:

Author: Thanks, fixed in https://github.com/gatsbyjs/gatsby/pull/17351/commits/17c227247ac4d824e8258f377f8c3686873e2800

```jsx:title=src/components/layout.js
import "typeface-your-chosen-font"

> `````

Reviewer 2: too much backticks...

```suggestion
```
```
